Set oauth2 refresh token expiration setting

fosterseth opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
bug fix for issue #4710 and #3825 

Sets `REFRESH_TOKEN_EXPIRE_SECONDS` in OAUTH2_PROVIDER settings to 1 month
This needs to be set in order for cleartokens or cleanup_tokens to run properly.

Once cleartokens is run, this setting is the amount of time after an access token expires that we keep its refresh token in the database

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 7.0.0

```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

fosterseth committed:
Set REFRESH_TOKEN_EXPIRE_SECONDS

- Set OAUTH2 REFRESH_TOKEN_EXPIRE_SECONDS to 1 month
  (2628000 seconds)
- If not set, awx-manage cleartokens, or cleanup_tokens,
  will not work properly
- Once cleartokens is run, this setting is the amount of
  time after an access token expires that we keep its
  refresh token in the database

rooftopcellist commented:
@fosterseth to make this setting available in the API so it can be consumed in the UI, there are a couple more changes necessary:
* need to register the setting in `awx/api/conf.py` - https://github.com/ansible/awx/blob/b858001c8f841d49a8b87ecfbf2bf898539385bb/awx/api/conf.py#L40
* need to make it a valid key type for this special field type: https://github.com/ansible/awx/blob/d309acfddb76b1cddfef611f075697dd0894ab55/awx/api/fields.py#L83

The rest of the changes are UI related and are covered by this PR: https://github.com/ansible/awx/pull/4890

fosterseth committed:
Register default settings for OAUTH2_PROVIDER app

Grab AUTHORIZATION_CODE_EXPIRE_SECONDS from oauth2_settings
rather than hard code.

Add REFRESH_TOKEN_EXPIRE_SECONDS to valid_key_names
in OAuth2ProviderField class

rooftopcellist commented on awx/api/conf.py:
```diff
@@ -38,7 +38,8 @@
+             'AUTHORIZATION_CODE_EXPIRE_SECONDS': oauth2_settings.AUTHORIZATION_CODE_EXPIRE_SECONDS,
+             'REFRESH_TOKEN_EXPIRE_SECONDS': oauth2_settings.REFRESH_TOKEN_EXPIRE_SECONDS},
     label=_('OAuth 2 Timeout Settings'),
     help_text=_('Dictionary for customizing OAuth 2 timeouts, available items are '
```
@fosterseth We should probably update this help text as well.

fosterseth committed:
update help_text to include information about REFRESH_TOKEN_EXPIRE_SECONDS

rooftopcellist reviewed (approved):
lgtm.  I will fold this commit in to my PR for the related UI changes.
